apps/token/src/routes/staking/staking-form.tsx

```diff
@@ -0,0 +1,319 @@
+import { useNetworkParam } from '../../hooks/use-network-param';
+import { useSearchParams } from '../../hooks/use-search-params';
+import { BigNumber } from '../../lib/bignumber';
+import { addDecimal, removeDecimal } from '../../lib/decimals';
+import type {
+  DelegateSubmissionInput,
+  UndelegateSubmissionInput,
+} from '../../lib/vega-wallet/vega-wallet-service';
+import { vegaWalletService } from '../../lib/vega-wallet/vega-wallet-service';
+import type {
+  PartyDelegations,
+  PartyDelegationsVariables,
+} from './__generated__/PartyDelegations';
+import { StakeFailure } from './stake-failure';
+import { StakePending } from './stake-pending';
+import { StakeSuccess } from './stake-success';
+import { Button, FormGroup } from '@vegaprotocol/ui-toolkit';
+
+export const PARTY_DELEGATIONS_QUERY = gql`
+  query PartyDelegations($partyId: ID!) {
+    party(id: $partyId) {
+      id
+      delegations {
+        amount
+        amountFormatted @client
+        node {
+          id
+        }
+        epoch
+      }
+    }
+    epoch {
+      id
+    }
+  }
+`;
+
+enum FormState {
+  Default,
+  Pending,
+  Success,
+  Failure,
+}
+
+export enum Actions {
+  Add = 'Add',
+  Remove = 'Remove',
+}
+export type StakeAction = Actions | undefined;
+export enum RemoveType {
+  EndOfEpoch,
+  Now,
+}
+
+interface StakingFormProps {
+  nodeId: string;
+  pubkey: string;
+  nodeName: string;
+  availableStakeToAdd: BigNumber;
+  availableStakeToRemove: BigNumber;
+}
+
+export const StakingForm = ({
+  nodeId,
+  pubkey,
+  nodeName,
+  availableStakeToAdd,
+  availableStakeToRemove,
+}: StakingFormProps) => {
+  const params = useSearchParams();
+  const navigate = useNavigate();
+  const client = useApolloClient();
+  const { appState } = useAppState();
+  const [formState, setFormState] = React.useState(FormState.Default);
+  const { t } = useTranslation();
+  const [action, setAction] = React.useState<StakeAction>(params.action);
+  const [amount, setAmount] = React.useState('');
+  const [removeType, setRemoveType] = React.useState<RemoveType>(
+    RemoveType.EndOfEpoch
+  );
+  // Clear the amount when the staking method changes
+  React.useEffect(() => {
+    setAmount('');
+  }, [action, setAmount]);
+  const { data } = useNetworkParam([
+    NetworkParams.VALIDATOR_DELEGATION_MIN_AMOUNT,
+  ]);
+  const minTokensWithDecimals = React.useMemo(() => {
+    const minTokens = new BigNumber(data && data.length === 1 ? data[0] : '');
+    return addDecimal(minTokens, appState.decimals);
+  }, [appState.decimals, data]);
+
+  const maxDelegation = React.useMemo(() => {
+    if (action === Actions.Add) {
+      return availableStakeToAdd;
+    }
+
+    if (action === Actions.Remove) {
+      return availableStakeToRemove;
+    }
+
+    return new BigNumber(0);
+  }, [action, availableStakeToAdd, availableStakeToRemove]);
+
+  // TODO: TFE import use useVegaTransaction hook from wallet lib
```

Comment:
Remove and add ticket for this

Reply:
https://github.com/vegaprotocol/frontend-monorepo/issues/270